core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -921,7 +921,10 @@ private[group] class GroupCoordinator(
         ): Unit = {
           val (error, verificationGuard) = errorAndGuard
           if (error != Errors.NONE) {
-            val finalError = GroupMetadataManager.maybeConvertOffsetCommitError(error)
+            val finalError = error match {
+              case Errors.NETWORK_EXCEPTION => Errors.COORDINATOR_LOAD_IN_PROGRESS
```

Comment:
I'm curious why we singled out this error explicitly rather than putting it into the function tasked specifically with error translation?

Reply:
Should we add a comment why we're translating to this error and not the other?

Reply:
Thank you for the review!

`maybeConvertOffsetCommitError` is also used in [`GroupMetadataManager.createPutCacheCallback`](https://github.com/apache/kafka/blob/7e85d7d32e8cb4f3fa9b01fd4197ee43d64ba6d0/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L430). I wasn't sure whether it made sense to translate network exceptions on that path too. I'm happy to move the translation there if you think it makes sense.

I have the same uncertainty about [`handleOperationException`](https://github.com/apache/kafka/blob/7e85d7d32e8cb4f3fa9b01fd4197ee43d64ba6d0/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java#L1084) in the new group coordinator. All the other group coordinator methods use it to translate errors and I was not sure if it made sense to translate network exceptions there.

Also, a comment to explain the choice of error would be good. I'll add one here and in ReplicaManager too.

Reply:
I've added a comment and moved the error translation in both group coordinators now.